### PR TITLE
Direct Tipping

### PR DESCRIPTION
**Pull request by thepiwo**

### CODE REVIEW

File: contracts/Tipping_v2_Standalone.aes [hunk 1]
```diff
@@ -110,25 +114,45 @@ contract Tipping =
...
+                         , amount = token_amount }
     let updated_token_claims = token_claims{ [token = 0] @ previous_amount = previous_amount + token_amount }
-    put(state{ tips[tip_id] = tip,
+    put(state{ tips[tip_id] = TokenTip(tip_meta, url_id, tip_token_data, last_claim_gen + 1),
                claims[url_id] = (last_claim_gen, amount, updated_token_claims) })
     token.transfer_allowance(Call.caller, Contract.address, token_amount)
     Chain.event(TipTokenReceived(Call.caller, token_amount, url, token))
     tip_id
 
+  payable stateful entrypoint tip_direct(receiver: receiver, title : string) : int =
```
Comment (mradkov) [inline contracts/Tipping_v2_Standalone.aes:129]: Should we change `title : string` to `title : option(string)` 🤔 

As most likely the direct tip won't require title/message to be passed along with the tip value.

Comment (thepiwo) [reply contracts/Tipping_v2_Standalone.aes:129]: Eventually empty string is also fine for this, let's discuss this further

File: util/tippingContractUtil.js [hunk 2]
```diff
@@ -28,11 +28,47 @@ tippingContractUtil.getTipsRetips = (state) => {
...
+        break;
+    }
+
+    data.type = data.type ? data.type : 'AeTip';
     data.id = id;
-    data.url = findUrl(data.url_id);
-    data.retips = findRetips(id, data.url_id);
-    data.claim = findClaimGen(data.claim_gen, data.url_id);
+
+    const hasClaim = data.claim_gen !== undefined && data.url_id !== undefined;
```
Comment (kenodressel) [inline util/tippingContractUtil.js:68]: can we rename this to something like isDirectTip?

**Review by kenodressel — APPROVED** [review]

File: util/tippingContractUtil.js (hunk 2 above)
Comment (thepiwo) [reply util/tippingContractUtil.js:68]: technically it decides if it has a claim, also this is just used internally to the util and is not exposed, so I think its fine as is

**Review by mradkov — APPROVED** [review]

**Pull request merged**